File: by-kotlin/kt-boj/solutions/2212-센서.kt
import java.util.*

fun main() = with(Scanner(System.`in`)) {
    val n = nextInt()
    val k = nextInt()

    if (k >= n) {
        print(0)
        return@with
    }

    val data = arrayListOf<Int>()
    repeat(n) { data.add(nextInt()) }

    data.sort()

    val distances = arrayListOf<Int>()
    for (i in 1 until n) {
        distances.add(data[i] - data[i - 1])
    }
    distances.sortDescending()

    for (i in 0 until k - 1) {
        distances[i] = 0
    }

    print(distances.sum())
}

File: by-python/boj/2212-센서.py
import sys

n = int(input())
k = int(input())

if k >= n:
    print(0)
    sys.exit()

data = list(map(int, input().split()))
data.sort()

distances = []
for i in range(1, n):
    distances.append(data[i] - data[i - 1])
distances.sort(reverse=True)

for i in range(k - 1):
    distances[i] = 0
print(sum(distances))
